docs/source/plugins.rst

```diff
@@ -0,0 +1,66 @@
+=======
+Plugins
+=======
+
+Plugins are extensions of rez that extend rez functionality.
+
+Plugins are currently bundled in the main rez repo, but will be split out
+to their own repos in the future.
+
+The plugins that exist are currently located at :gh-rez:`src/rezplugins`.
+
+Existing plugins
+================
+
+- :gh-rez:`src/rezplugins/build_process`
+- :gh-rez:`src/rezplugins/build_system`
+- :gh-rez:`src/rezplugins/command`
+- :gh-rez:`src/rezplugins/build_process/package_repository`
+- :gh-rez:`src/rezplugins/build_process/release_hook`
+- :gh-rez:`src/rezplugins/build_process/release_vcs`
+- :gh-rez:`src/rezplugins/build_process/shell`
+
+Developing your own plugin
+==========================
+
+Rez plugins require a specific folder structure as follows:
+
+.. code-block:: text
+
+    /plugin_name
+        /__init__.py (adds plugin path to rez)
+        /rezconfig.py (defines configuration settings for your plugin)
+        /plugin_file1.py (your plugin file)
+        /plugin_file2.py (your plugin file)
+        etc.
+
+To make your plugin available to rez, you can install them directly under
+``src/rezplugins`` (that's called a namespace package) or you can add
+the path to :envvar:`REZ_PLUGIN_PATH`.
+
+Plugin types
+------------
+
+There are two different plugin types in Rez.
+
+- Plugin
+- Extension
+
+Extensions differ from standard plugins in that they add a new command to rez's
+CLI.
```

Comment:
I'm not too sure about this section. "extensions" are not special from what I know. We could explain what an extension is by saying that a plugin that adds a command is sometimes refereed to as an extension?

Reply:
Yeah I would drop "extension" entirely if possible. Plugins exist. You can do a lots of things with plugins, including adding `rez` commands. In fact, it's impossible to add a plugin without "extending" rez's functionality in some way so calling only plugins that add `rez` commands "extensions" is arguably inaccurate, or incorrect, by some folks' definition.